Add notifications controller and modal

kleyv opened the pull request:
Removed Javascript to toggle the bell and replaced the logic
with actions to create and destroy notifications.
Added self methods to the Trip class to check for notifications
and get notifications ids.
Gerenated a notifications controller and routes.
Added notifcation_modal with placeholder for future form.


Work in progress **do not merge**. Still have to figure out how to get the trip and notification ids on the modal.

kleyv committed:
Add notifications controller and modal

Removed Javascript to toggle the bell and replaced the logic
with actions to create and destroy notifications.
Added self methods to the Trip class to check for notifications
and get notifications ids.
Gerenated a notifications controller and routes.
Added notifcation_modal with placeholder for future form.

kleyv committed:
Merge branch 'master' into add-case-notifications

kleyv committed:
Get the threshold value from the form and add it to the model

Modified the notification destroy method to also reset the
new_daily_cases_thresholds number on the trip to zero when
deleting the notification. And changed the notice to an alert
instead.
Removed last remains of previous Javascript code for the notification
bell.
Moved the notification_modal to the Trips views folder.
Modified the route so that get_notified would act as a patch on
the trip model instead of a post on the notification model.

kleyv committed:
Explain the @trip variable on Trips index view

kleyv committed:
Fix typo in a Trips column and add notify column

kleyv committed:
Change logic to activate notifications instead of creating them

Removed notifcations destroy method and mode its logic to trips
disable_notifications method.
Make it so the click of the bell subscribes the user to
notifications insteead of immediately creating a notification.
Fixed the last remnants of new_daily_cases_threshold(s) typo.

kleyv committed:
Add feature to edit notification

kleyv committed:
Add job to create notifications as needed

kleyv committed:
Solve conflicts

kleyv committed:
Add notifications index action and view

kleyv committed:
Update notication content and create Twilio job

Added new_daily_cases_notification method to twilio_client service
to send an sms alerting the user to new cases.
Added CreateNotificationJob to case:update task.

Flott2410 commented:
Very good work. Sum bugs occured. See my comments in Slack. I wait for the merge. May you can check again or we do a session together Monday to work on it together. Maybe we can meet at some place in Lisbon. E.g. DeBrew Cowork

kleyv committed:
Move notification actions to trips show page

Moved notification activation and destruction to trips show page
because it's easier to hvae access to a trip id and manipulate it.
Commented out the Notifications hyperlink from the navbar leaving
the view intact. For we'll focus only on SMS notifications.
